scripts/compactor_utils.py

```diff
@@ -0,0 +1,55 @@
+import logging
+import subprocess
+import time
+import threading
+
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+
+def set_logger(config_file_path):
+    with open(config_file_path, "r") as config:
+        compactor_config = yaml.load(config)
+    corfu_paths = compactor_config["CorfuPaths"]
+    logging.basicConfig(filename=corfu_paths["CompactorLogfile"],
+                        format='%(asctime)s.%(msecs)03dZ %(levelname)5s Runner - %(message)s',
+                        datefmt='%Y-%m-%dT%H:%M:%S')
+
+def log_info(msg):
+    logger.info(msg)
+
+def wait_for_compactor_end(port, target_cycle_count):
+    while True:
+        status, cycle_count = get_compactor_status(port)
+        if stop_threads is True or cycle_count >= target_cycle_count and status != "STARTED":
+            log_info("status: " + status + " cycleCount: " + str(cycle_count))
+            return
+        else:
+            time.sleep(5)
+
+def get_compactor_status(port):
+    cmd = "/opt/vmware/bin/corfu_tool_runner.py -t CompactionManagerTable -n CorfuSystem -o showTable --port " + port
+    result = subprocess.check_output(cmd, shell=True).decode()
+    status = "IDLE"
+    cycle_count = 0
+    for line in result.split("\n"):
+        if "status\"" in line:
+            status = line.split(":")[-1].split("\"")[1]
+        if "cycleCount\"" in line:
+            cycle_count = line.split(":")[-1].split("\"")[1]
+            break
+    return status, int(cycle_count)
+
+def wait_and_verify_compactor_success(port, target_cycle_count):
+    compaction_status_thread = threading.Thread(target=wait_for_compactor_end, args=(port, target_cycle_count))
+    global stop_threads
+    stop_threads = False
+    compaction_status_thread.start()
+    compaction_status_thread.join(timeout=600)
+    if compaction_status_thread.is_alive():
+        stop_threads = True
```

Comment:
the `stop_threads` won't take effect because there is a join on line 47 right? Am I missing something?
Instead of using a global variable do you want the `wait_for_compactor_end` itself to give up after a fixed number of retries

Reply:
Yes I think that's safer. Will update.